src/entity/category/category_id.go

```diff
@@ -0,0 +1,14 @@
+package category
+
+import custerr "github.com/ha-zu/learn-clean-dev/src/entity/customerror"
+
+type CategoryID string
+
+func NewCategoryID(id CategoryID) error {
+
+	if id == "" {
+		return custerr.ErrEmptyValue
+	}
+
+	return nil
+}
```

Comment:
```suggestion
func NewCategoryID() CategoryID {

	return CategoryID(uuid.New().String())
}
```

こんな感じでIDの作成自体の責務をvalue object側に持っておいてOKです。
またuuidのパッケージは"github.com/google/uuid"を使うと良いです！


後stringからCategoryIDに変換する用の関数もあったほうが良いです。
それが既に書かれているコードと似ていて以下のような感じになります
```suggestion
func NewCategoryIDByStr(id string) (CategoryID, error) {

	if id == "" {
		return CategoryID(""), custerr.ErrEmptyValue
	}

	return CategoryID(id), nil
}
```

他のID全て同様です！